### PR TITLE
Kotlin Cleanup CardTemplateEditor

### PR DESCRIPTION
**Pull request by dorrin-sot**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 1]
```diff
@@ -974,5 +969,11 @@ open class CardTemplateEditor : AnkiActivity(), DeckSelectionListener {
         private const val REQUEST_PREVIEWER = 0
         @Suppress("unused")
         private const val REQUEST_CARD_BROWSER_APPEARANCE = 1
+
+        inline fun <reified K, reified V> Serializable?.toHashMap(): HashMap<K, V>? {
+            return (this as? HashMap<*, *>)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt]: Your method hides a potential ClassCastException because of `as?`, after invoking your `toHashMap()` I don't know if there was a null value put as the HashMap or a ClassCastException was thrown.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 2]
```diff
@@ -112,13 +111,14 @@ open class CardTemplateEditor : AnkiActivity(), DeckSelectionListener {
...
-            mStartingOrdId = savedInstanceState.getInt(EDITOR_START_ORD_ID)
-            tabToCursorPosition = savedInstanceState.getSerializable(TAB_TO_CURSOR_POSITION_KEY) as HashMap<Int, Int?>?
-            tabToViewId = savedInstanceState.getSerializable(TAB_TO_VIEW_ID) as HashMap<Int, Int?>?
-            tempModel = TemporaryModel.fromBundle(savedInstanceState)
+        }
+        savedInstanceState?.run {
+            mModelId = getLong(EDITOR_MODEL_ID)
+            mNoteId = getLong(EDITOR_NOTE_ID)
+            mStartingOrdId = getInt(EDITOR_START_ORD_ID)
+            tabToCursorPosition = getSerializable(TAB_TO_CURSOR_POSITION_KEY).toHashMap()
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:124]: We already have the BundleUtils.getSerializableWithCast() that you can use.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 3]
```diff
@@ -158,33 +158,29 @@ open class CardTemplateEditor : AnkiActivity(), DeckSelectionListener {
...
         // Close collection opening dialog if needed
         Timber.i("CardTemplateEditor:: Card template editor successfully started for model id %d", mModelId)
 
         // Set the tab to the current template if an ord id was provided
-        Timber.d("Setting starting tab to %d", mStartingOrdId)
-        if (mStartingOrdId != -1) {
-            viewPager.setCurrentItem(mStartingOrdId, animationDisabled())
-        }
+        mStartingOrdId
+            .takeUnless { it == -1 }
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:182]: There is no need for this change, the previous code reads more cleanly. It also doesn't log with Timber every time like the previous code.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 4]
```diff
@@ -40,10 +40,13 @@ import com.afollestad.materialdialogs.MaterialDialog
 import com.google.android.material.bottomnavigation.BottomNavigationView
 import com.google.android.material.tabs.TabLayout
 import com.google.android.material.tabs.TabLayoutMediator
-import com.ichi2.anim.ActivityTransitionAnimation.Direction.*
-import com.ichi2.anki.dialogs.*
+import com.ichi2.anim.ActivityTransitionAnimation.Direction.END
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:43]: Try to keep commits as simple as possible without unrelated changes.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 5]
```diff
@@ -185,7 +188,7 @@ open class CardTemplateEditor : AnkiActivity(), DeckSelectionListener {
 
     fun modelHasChanged(): Boolean {
         val oldModel: JSONObject? = col.models.get(mModelId)
-        return tempModel != null && tempModel!!.model.toString() != oldModel.toString()
+        return tempModel?.model.toString() !in listOf("null", oldModel.toString())
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:191]: There's no need for this change, the previous code reads more cleanly.

**Comment by lukstbit** [comment]
Thank you for your contributions, I've reviewed some of the commits.
I would ask you to split this PR in smaller PRs which are tied together(for example the changes in TemplatePagerAdapter in one PR, multiple changes in the same method in another PR). This would make it easier to review, avoid merge conflicts and get more attention to your PRs.

**Comment by mikehardy** [comment]
We have all been focused on other things (I have definitely been focused elsewhere at least...) but this still has potential
